dblocal/memory_storage.go

```diff
@@ -0,0 +1,89 @@
+package dblocal
```

Comment:
It would be much prettier if this resided in the db package alongside the current leveldb implementation. There could be a "main" file which exposes either one of these with a flag.